### PR TITLE
[T2A2][T15-A2]Amro Shohoud

### PR DESCRIPTION
**Pull request by AmroShohoud**

### CODE REVIEW

**Comment by AmroShohoud** [comment]
Ready for review

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -262,11 +262,11 @@ private static void processProgramArgs(String[] args) {
             exitProgram();
         }
 
-        if (args.length == 1) {
+        if (args.length == 1) { //File given
```
Comment (xpdavid) [inline src/seedu/addressbook/AddressBook.java:265]: It may be a unnecessary comment here.

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -394,8 +394,9 @@ private static String executeCommand(String userInputString) {
      * @return  size 2 array; first element is the command type and second element is the arguments string
      */
     private static String[] splitCommandWordAndArgs(String rawUserInput) {
-        final String[] split =  rawUserInput.trim().split("\\s+", 2);
-        return split.length == 2 ? split : new String[] { split[0] , "" }; // else case: no parameters
+    	int splittedElements = 2;
```
Comment (xpdavid) [inline src/seedu/addressbook/AddressBook.java:397]: Note the indentation here. Should be 4 spaces.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -394,8 +394,9 @@ private static String executeCommand(String userInputString) {
...
+        final String[] splittedCommand =  rawUserInput.trim().split("\\s+", splittedElements);
+        return splittedCommand.length == splittedElements ? splittedCommand : new String[] { splittedCommand[0] , "" }; // else case: no parameters
```
Comment (xpdavid) [inline src/seedu/addressbook/AddressBook.java:399]: For long ternary expressions, you may want to refer to the format in coding standard.

File: src/seedu/addressbook/AddressBook.java (hunk 2 above)
Comment (xpdavid) [reply src/seedu/addressbook/AddressBook.java:397]: Good job! Removing magic number increase the readability of code.

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -1143,17 +1138,24 @@ private static String getUsageInfoForExitCommand() {
      * ============================
      */
 
+    
     /**
      * Removes sign(p/, d/, etc) from parameter string
      *
      * @param s  Parameter as a string
      * @param sign  Parameter sign to be removed
```
Comment (xpdavid) [inline src/seedu/addressbook/AddressBook.java:1146]: When you update method signature, you also need to update javadoc

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -1143,17 +1138,24 @@ private static String getUsageInfoForExitCommand() {
...
      * @return  string without the sign
      */
-    private static String removePrefixSign(String s, String sign) {
-        return s.replace(sign, "");
+    private static String removePrefix(String fullString, String prefix) {
+    	if (fullString.indexOf(prefix) == 0) {
+    		return fullString.replace(prefix,  "");
+    	}
+    	else {
```
Comment (xpdavid) [inline src/seedu/addressbook/AddressBook.java:1153]: Note that there is coding standard violation here.

File: src/seedu/addressbook/AddressBook.java [hunk 6]
```diff
@@ -557,14 +558,15 @@ private static boolean isDisplayIndexValidForLastPersonListingView(int index) {
     private static String getMessageForSuccessfulDelete(String[] deletedPerson) {
         return String.format(MESSAGE_DELETE_PERSON_SUCCESS, getMessageForFormattedPersonData(deletedPerson));
     }
-
+
```
Comment (xpdavid) [inline src/seedu/addressbook/AddressBook.java:561]: This is unnecessary change.

**Comment by xpdavid** [comment]
@AmroShohoud Some comments added. Please close the PR after reading comments.

Remember to update your PR title to the one that match your group.
